### PR TITLE
updates/upgrades for hauler `v1.0.0`

### PR DESCRIPTION
**Pull request by zackbradys**
### code related changes:
- Upgraded to `docusaurus v3`, specifically `docusaurus v3.1.1`
- Upgraded to `docusaurus-lunr-search` (from `docusaurus-search-local`) due to `docusaurus v3`
  - https://github.com/praveenn77/docusaurus-lunr-search
- Modified previous versions of the docs to support breaking changes in `docusaurus v3`
  - https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3

### docs related changes:
- Updated the docs to support new features and functionalities in `hauler v1.0.0`
- Updated existing pages in the docs based on feedback from users and recent demo
- Updated content in existing pages that have been "pending release"

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zackbradys** [comment]
Preview the updates here... http://54.235.23.85:3000/hauler-docs

**Comment by zackbradys** [comment]
Preview site updated and ready for review!

File: docs/guides-references/hauler-content/charts.md [hunk 1]
```diff
@@ -1,24 +1,54 @@
...
 
-Hauler supports the experimental format of Helm Chart OCI Artifacts, but can also be represented as the usual Tarball Artifact!
-
-:::
+```bash
+hauler store add chart <chart name> --repo <chart repository> --version <chart version>
+```
 
-Helm charts represented as OCI content:
+## Example Command for Images
```
Comment (amartin120) [inline docs/guides-references/hauler-content/charts.md:13]:
```suggestion
## Example Commands for Charts
```

File: docs/guides-references/hauler-content/files.md [hunk 2]
```diff
@@ -1,15 +1,55 @@
...
+## Hauler Command Line for Files
 
-```yaml
-# local file
-hauler store add file path/to/local/file.txt
+```bash
+hauler store add file <file> --name <name>
+```
+
+## Example Command for Images
```
Comment (amartin120) [inline docs/guides-references/hauler-content/files.md:13]:
```suggestion
## Example Command for Files
```

File: docs/guides-references/hauler-content/files.md [hunk 3]
```diff
@@ -1,15 +1,55 @@
...
+
+```bash
+# fetch helm chart
+hauler store add file https://get.rke2.io
```
Comment (amartin120) [inline docs/guides-references/hauler-content/files.md:17]:
```suggestion
# local file
hauler store add file path/to/local/file.txt
```

File: docs/guides-references/hauler-content/files.md [hunk 4]
```diff
@@ -1,15 +1,55 @@
...
 
-# remote file with name
+# fetch helm chart with specific version
```
Comment (amartin120) [inline docs/guides-references/hauler-content/files.md:19]:
```suggestion
# remote file with name
```

**Review by amartin120 — CHANGES_REQUESTED** [review]

**Comment by amartin120** [comment]
Looks great!  Just a few suggestions.

**Comment by zackbradys** [comment]
> Looks great! Just a few suggestions.

Updated and terminated the preview site.

**Pull request merged**